### PR TITLE
229 - Basic home page with the list of datasets

### LINKED ISSUE #229: Basic Home Page with the list of datasets
## Overview of the Feature Request
Add a new page to the SPA to serve as the Home Page. This page will include the list of the datasets that are available to the user. 

Functionalities:
- List of datasets titles with links to the corresponding dataset page (no files or collections)
- Pagination (10 datasets per page)

## Mock
Captura de pantalla 2023-11-15 a las 10 52 52

## What kind of user is the feature intended for?
All of them

## What inspired the request?
Frontend meeting discussion about the next user workflows that could extend the SPA to cover more features horizontally without too much detail.

## What existing behavior do you want changed?
None

## Any brand new behavior do you want to add to Dataverse?
No

### PR DESCRIPTION
**Pull request by MellyGray**
## What this PR does / why we need it:
This PR adds a basic home page to the SPA to display a list of available datasets to the user. Note that this is only the frontend so is not connected with the backend yet.

## Which issue(s) this PR closes:

- Closes #229

## Special notes for your reviewer:
I refactored FilePaginationInfo class to extract a common class between FilePaginationInfo and DatasetPaginationInfo, this way we can rehuse the pagination functionality.

I'm aware that the Dataverse home page shows collections, datasets and files, but for the aim of simplicity in this first iteration only the datasets are being displayed.

## Suggestions on how to test this:
1. Install the necessary dependencies using `npm install`.
5. Build the design system with `cd packages/design-system && npm run build`.
6. Navigate back to the root directory with `cd ../../`.
7. Start Storybook with `npm run storybook`.
8. Visit [the Home in the Storybook](http://localhost:6006/?path=/story/pages-home--default)
9. For detailed stories of the Home components you can go to the [Home section of the storybook](http://localhost:6006/?path=/story/sections-home-datasetslist--default) 

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
Captura de pantalla 2023-11-15 a las 10 52 52

## Is there a release notes update needed for this change?:
Datasets list added to the SPA home page

### CODE REVIEW

**Comment by GPortas** [comment]
@MellyGray Please, can you resolve conflicts?

File: src/dataset/domain/models/DatasetPaginationInfo.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { PaginationInfo } from '../../../shared/domain/models/PaginationInfo'
+
+export class DatasetPaginationInfo extends PaginationInfo<DatasetPaginationInfo> {
+  constructor(page = 1, pageSize = 10, totalItems = 0, itemName = 'Dataset') {
```
Comment (GPortas) [inline src/dataset/domain/models/DatasetPaginationInfo.ts:4]: I understand that the values are intentionally hardcoded and that in future iterations pagination will be introduced as in files tab?

**Review by GPortas — APPROVED** [review]
Code looks good.

The UI looks good (Understanding that it is a boilerplate to build on):

rootdatasete

**Comment by MellyGray** [comment]
@GPortas I know the boilerplate UI is kind of ugly, we could merge this other PR here so the ugly UI doesn't make it into beta:
- https://github.com/IQSS/dataverse-frontend/pull/248

**Comment by GPortas** [comment]
Waiting for https://github.com/IQSS/dataverse-frontend/pull/248 to be merged

**Comment by GPortas** [comment]
@MellyGray Can you please resolve the conflicts?

**Comment by MellyGray** [comment]
@GPortas conflicts resolved

**Comment by pdurbin** [comment]
Now that #248 has been merged into this PR, can we remove the "waiting" label?

**Review by GPortas — APPROVED** [review]
Looks good!

datasetshome

**Pull request merged**